modules/passkey/test/userstories/ExecuteWithPaymaster.spec.ts

```diff
@@ -0,0 +1,371 @@
+        [paymaster.target, paymasterVerificationGasLimit, paymasterPostOpGasLimit, paymasterData],
+      )
+
+      packedUserOp.paymasterAndData = paymasterAndData
+
+      // Step 4: Sign userOp with Passkey signer.
+      // opHash that will be signed using Passkey credentials
+      const opHash = await module.getOperationHash(packedUserOp)
+
+      const assertion = navigator.credentials.get({
+        publicKey: {
+          challenge: ethers.getBytes(opHash),
+          rpId: 'safe.global',
+          allowCredentials: [{ type: 'public-key', id: new Uint8Array(credential.rawId) }],
+          userVerification: 'required',
+        },
+      })
+
+      // Build the contract signature that a Safe will forward to the signer contract
+      const signature = buildSignatureBytes([
+        {
+          signer: signer,
+          data: encodeWebAuthnSignature(assertion.response),
+          dynamic: true,
+        },
+      ])
+
+      // Set the signature in the packedUserOp
+      packedUserOp.signature = ethers.solidityPacked(['uint48', 'uint48', 'bytes'], [safeOp.validAfter, safeOp.validUntil, signature])
+
+      // Check if Safe is not already created
+      expect(await ethers.provider.getCode(safe)).to.equal('0x')
+
+      // Step 5: Execute the userOp that deploys a Safe with passkey signer as owner.
+      await entryPoint.handleOps([packedUserOp], relayer.address)
+
+      // Check if Safe is created and uses the expected Singleton
+      const [implementation] = ethers.AbiCoder.defaultAbiCoder().decode(['address'], await ethers.provider.getStorage(safe, 0))
+      expect(implementation).to.equal(singleton.target)
+
+      // Check if signer is the Safe owner
+      const safeInstance = await ethers.getContractAt(SafeL2.abi, safe)
+      expect(await safeInstance.getOwners()).to.deep.equal([signer])
+    })
+  })
+
+  /**
+   * The flow can be summarized as follows:
+   * Step 1: Setup the contracts.
+   * Step 2: Create a userOp with callData that transfers 0.2 ethers to address(0).
+   * Step 3: Get signature from verifyingSigner account and create paymaster data.
+   * Step 4: Sign userOp with Passkey signer.
+   * Step 5: Execute the userOp that with an existing Safe with passkey signer as owner.
+   */
+  describe('Existing Safe', () => {
+    // Create a fixture to setup the contracts and signer(s)
+    const setupTests = deployments.createFixture(async ({ deployments }) => {
+      const {
+        relayer,
+        verifyingSigner,
+        proxyFactory,
+        safeModuleSetup,
+        module,
+        entryPoint,
+        singleton,
+        verifier,
+        credential,
+        paymaster,
+        signerFactory,
+        navigator,
+      } = await generalSetup(deployments)
+
+      const publicKey = decodePublicKey(credential.response)
+      await signerFactory.createSigner(publicKey.x, publicKey.y, verifier.target)
+      const signer = await signerFactory.getSigner(publicKey.x, publicKey.y, verifier.target)
+
```

Comment:
seems like both tests deploy a signer right after the setup, can it be a part of the general setup?

Reply:
Agree it might be nice (so the tests assume an existing signer).